src/browser/chrome.ts

```diff
@@ -40,16 +40,49 @@ export const closeBrowser = async () => {
 };
 
 const WINDOW_VARIABLE_NAME = `___webrender`;
+const LAST_RESPONSE_AT_VARIABLE_NAME = `__webrenderLastResponseAt`;
+const COUNTER_REQUEST_UNFINISHED_VARIABLE_NAME = `__webrenderCounterReqUnfinished`;
 export const getPageInitScriptFor = (
   js: string,
-  jsOn: JsOn = "commit"
-) => `try {
-  const AsyncFunction = Object.getPrototypeOf(
+  jsOn: JsOn = "commit",
+  timeout: number
+) => {
+  const PAGE_TRUE_LOAD = `
+  async pageTrueLoad ({
+    idleTimeout = 2000, resolveBeforeRenderingTimeout = 2000
+  } = {}) {
+    const timeoutPromise = new Promise((resolve) => {
+      setTimeout(() => {
+        resolve();
+      }, ${timeout} - resolveBeforeRenderingTimeout);
```

Comment:
`timeout` var won't work reliably here. You need something like `timeoutAt` (absolute time) passed from the higher context.

say the api user requested `timeout=30000`

By the time this code is invoked, you [could have already spent](https://github.com/nikitaeverywhere/webrender/blob/9f91782732e20479d0f05bf3c607a67303f546e6/src/browser/chrome.ts#L179) for instance 4 seconds navigating to the page etc. 

Please ensure this timeout is set relatively to the absolute rendering end timestamp.